src/platform/nrfconnect/FactoryDataParser.c

```diff
@@ -58,7 +71,11 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
             break;
         }
 
-        if (strncmp("hw_ver", (const char *) currentString.value, currentString.len) == 0)
+        if (strncmp("version", (const char *) currentString.value, currentString.len) == 0)
```

Comment:
hmm, how about using `uint16` for version? It won't change much in terms of size and will greatly reduce the chance that it will be insufficient.